modules/core/src/main/java/org/apache/ignite/internal/processors/cache/persistence/GridCacheDatabaseSharedManager.java

```diff
@@ -1656,8 +1656,14 @@ private String cacheInfo(GridCacheContext cacheCtx) {
 
                 int partId = e0.getKey();
 
-                assert cctx.wal().reserved(cpEntry.checkpointMark())
-                    : "WAL segment for checkpoint " + cpEntry + " has not reserved";
+                if (reservedCheckpointMark != null && !cctx.wal().reserved(reservedCheckpointMark)) {
+                    log.warning("Reservation failed because the segment was released: " + reservedCheckpointMark);
```

Comment:
Well, the `reservedCheckpointMark` is not changed after initialization at line 1639.
However, if I am not mistaken, it can be "released" at any time by https://issues.apache.org/jira/browse/IGNITE-14952 Perhaps, I am wrong, and we assume that this method `reserveHistoryForExchange` is called from the exchange thread (GridDhtPartitionsExchangeFuture#distributedExchange) and so, WAL segments cannot be released at this point.

Anyway, let's assume that `reservedCheckpointMark` was released right after its initialization. In that case, the warning message will be printed for all cache groups that were applicable for historical rebalance. Maybe I am missing something...

Reply:
In IGNITE-14952, we assume that the forced release of segments will occur only when the WAL of the archive is exceeded, which can only happen when segments are added to it.

Reply:
The question was removed, discussed in person.